modules/phase_field/src/userobjects/DiscreteNucleationInserter.C

```diff
@@ -24,15 +24,28 @@ validParams<DiscreteNucleationInserter>()
   params.addRequiredParam<MaterialPropertyName>(
       "probability", "Probability density for inserting a discrete nucleus");
   params.addRequiredParam<Real>("hold_time", "Time to keep each nucleus active");
+
+  params.addParam<bool>("fixed_radius", true, "flag if the radius is fixed or not");
+  params.addParam<Real>("radius", 0.0, "fixed radius (if using)");
+  params.addParam<MaterialPropertyName>("radius_material", "r_crit", "variable radius material property name");
```

Comment:
No. Let's only accept a material property here. The user can simply specify a number in place of the property. MOOSE does that for us. No need to implement our own logic.